Framework/API/inc/MantidAPI/IndexTypeProperty.h

```diff
@@ -0,0 +1,61 @@
+#ifndef MANTID_API_INDEXTYPEPROPERTY_H_
+#define MANTID_API_INDEXTYPEPROPERTY_H_
+
+#include "MantidAPI/DllConfig.h"
+#include <MantidKernel/PropertyWithValue.h>
+
+namespace Mantid {
+namespace API {
+
+enum IndexType { SpectrumNum = 1, WorkspaceIndex = 2 };
+
+/** IndexTypeProperty : TODO : Add support for DetectorIDs
```

Comment:
Missing TODO. 

Another minor point. I think the name `IndexTypeProperty` next to the `IndexProperty` are a little confusing. I'm not sure that `IndexTypeProperty` is a great name (but I can't think of anything better for now). One problem I have with it is that `SpectrumNumber` , for example, is not an `IndexType`